pages/api/books/PATCH.ts

```diff
@@ -0,0 +1,49 @@
+import prisma from '@/prisma/prisma';
+import matchRelations from './helperFunctions';
+
+type UpdateBody = {
+  barcodeId: string;
+  title: string;
+  author: string;
+  genres: string[];
+  regions: string[];
+  location: string;
+};
+
+export default async function updateHandler(id: string, body: UpdateBody) {
+  const { barcodeId, title, author, genres, regions, location } = body;
+
+  const [matchingGenres, matchingRegions] = await matchRelations(
```

Comment:
Extracted this out to helperFunctions.ts to keep things DRY between PATCH and POST.